app/Services/Model/Imp/ScheduleService.php

```diff
@@ -0,0 +1,262 @@
+
+            $selection_plan = $event->getSelectionPlan();
+            if (!$selection_plan instanceof SelectionPlan)
+                throw new EntityNotFoundException("event id {$presentation_id} does not have a selection plan");
+
+            if (!$selection_plan->isAllowProposedSchedules())
+                throw new ValidationException("selection plan id {$selection_plan->getId()} does not allow proposed schedules");
+
+            if (count($schedules) == 0) {
+                $member = ResourceServerContext::getCurrentUser(false);
+                $schedule = new SummitProposedSchedule();
+                $schedule->setSource($source);
+                $schedule->setSummit($event->getSummit());
+                if (isset($data['schedule_name'])) {
+                    $schedule->setName($data['schedule_name']);
+                } else {
+                    $schedule->setName("{$source} Proposed Schedule");
+                }
+                $schedule->setCreatedBy($member);
+                $this->schedule_repository->add($schedule);
+            } else {
+                $schedule = $schedules[0];
+            }
+            return $schedule;
+        });
+
+        return $this->publishProposedActivity($schedule->getId(), $presentation_id, $payload);
+    }
+
+    /**
+     *@inheritDoc
+     */
+    public function publishProposedActivity(
+        int $schedule_id, int $presentation_id, array $payload):SummitProposedScheduleSummitEvent {
+
+        return $this->tx_service->transaction(function () use ($schedule_id, $presentation_id, $payload) {
+
+            $schedule = $this->schedule_repository->getById($schedule_id);
+
+            if (!$schedule instanceof SummitProposedSchedule)
+                throw new EntityNotFoundException(sprintf("schedule id %s does not exists!", $schedule_id));
+
+            $event = $this->event_repository->getById($presentation_id);
+
+            if (!$event instanceof SummitEvent)
+                throw new EntityNotFoundException(sprintf("event id %s does not exists!", $presentation_id));
+
+            $member = ResourceServerContext::getCurrentUser(false);
+            $summit = $schedule->getSummit();
+
+            if (!$this->isAuthorizedUser( $member, $summit, $event->getCategory()))
+                throw new AuthzException("User is not authorized to perform this action");
+
+            $schedule_event = $schedule->getScheduledSummitEventByEvent($event);
+
+            if (is_null($schedule_event)) {
+                $default_date = new \DateTime("now", new \DateTimeZone("UTC"));
+                $schedule_event = new SummitProposedScheduleSummitEvent();
+                $schedule_event->setSummitEvent($event);
+                $schedule_event->setCreatedBy($member);
+                $schedule_event->setSchedule($schedule);
+                $schedule_event->setStartDate($event->getStartDate() ?? $default_date);
```

Comment:
lets remove this , we dont wanna to set up any default startdate/end date

Reply:
We need to have these values initialized bc of the further validation previous the initialization with the final values. This will be overridden immediately after that validation